docs/topics/tasks.txt

```diff
@@ -0,0 +1,337 @@
+background. This allows background task functionality to be slowly added
+to an application, before the required infrastructure is available.
+
+.. warning::
+
+    When :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>` is ``False``,
+    the task will be executed within the same transaction it was enqueued in.
+
+    This may lead to unexpected behavior changes when changing backend in future.
+
+To use it, set :setting:`BACKEND <TASKS-BACKEND>` to
+``"django.tasks.backends.immediate.ImmediateBackend"``::
+
+    TASKS = {"default": {"BACKEND": "django.tasks.backends.immediate.ImmediateBackend"}}
+
+.. _dummy-task-backend:
+
+Dummy backend
+-------------
+
+The dummy backend doesn't execute enqueued tasks at all, instead storing results
+for later use.
+
+This backend is not intended for use in production - it is provided as a
+convenience that can be used during development and testing.
+
+To use it, set :setting:`BACKEND <TASKS-BACKEND>` to
+``"django.tasks.backends.dummy.DummyBackend"``::
+
+    TASKS = {"default": {"BACKEND": "django.tasks.backends.dummy.DummyBackend"}}
+
+The results for enqueued tasks can be retrieved from the backend's
+:attr:`results <django.tasks.backends.dummy.DummyBackend.results>` attribute::
+
+    from django.tasks import default_task_backend
+
+    my_task.enqueue()
+
+    assert len(default_task_backend.results) == 1
+
+Stored results can be cleared using the
+:meth:`clear <django.tasks.backends.dummy.DummyBackend.clear>` method::
+
+    default_task_backend.clear()
+
+    assert len(default_task_backend.results) == 0
+
+Using a custom backend
+----------------------
+
+While Django includes support for a number of task backends out-of-the-box,
+sometimes you might want to customize the the task backend. To use an external
+task backend with Django, use the Python import path as the
+:setting:`BACKEND <TASKS-BACKEND>` of the :setting:`TASKS` setting, like so::
+
+    TASKS = {
+        "default": {
+            "BACKEND": "path.to.backend",
+        }
+    }
+
+If you're building your own backend, you can use the standard task backends
+as reference implementations. You'll find the code in the
+:source:`django/tasks/backends/` directory of the Django source.
+
+Backend arguments
+-----------------
+
+Each backend can be given additional arguments to control its behavior.
+These arguments are provided as additional keys in the :setting:`TASKS` setting.
+Valid arguments are as follows:
+
+* :setting:`ENQUEUE_ON_COMMIT <TASKS-ENQUEUE_ON_COMMIT>`: Whether a task should
+  be enqueued at the end of the current transaction (if there
+  is one) commits successfully, rather than enqueueing immediately. This argument
+  defaults to ``True``.
+
+* :setting:`QUEUES <TASKS-QUEUES>`: Restrict the queue names a task may be
+  enqueued to. By default, only the ``"default"`` queue is allowed. Queue name
+  validation can be disabled by setting this to an empty list.
```

Comment:
I am referring to the setting in general. Personally, I would not use this restriction. Not because I am expecting to dynamically create queues, but just that it feels like it would be Yet Another Thing To Remember To Do when adding tasks.

I don't feel strongly about this, though. If this feels like a useful safeguard to people, then it's a useful safeguard.

Reply:
The use case that I’ve encountered is when there are specific workers that must handle a particular task.

A real world example is pulling data from branch office databases. Suppose an instance of Postgres is running inside a corporate network in New York, and another instance of Postgres is running in London. Both of those sit inside private corporate networks, not publicly accessible.

To make these ETL pipelines work, we need a worker running in New York and one in London. The `refresh_newyork_data()` task must run on the New York worker, and the `refresh_london_data()` must run on the London worker.

Reply:
That's a convincing reason! Sounds good

Reply:
It would be useful to include this example in the docs, so people better understand why this is useful.